src/utils/eigen_utils.cpp

```diff
@@ -0,0 +1,50 @@
+#include "eigen_utils.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <numeric>
+
+Eigen::MatrixXd bayesmix::vstack(const std::vector<Eigen::MatrixXd> &mats) {
+  auto cnt_rows = [&](int curr, const Eigen::MatrixXd &mat) {
+    return curr + mat.rows();
+  };
+  int nrows = std::accumulate(mats.begin(), mats.end(), 0, cnt_rows);
+  int ncols = mats[0].cols();
+
+  Eigen::MatrixXd out(nrows, ncols);
+
+  int begin = 0;
+
+  for (int i = 0; i < mats.size(); i++) {
+    out.block(begin, 0, mats[i].rows(), ncols) = mats[i];
+    begin += mats[i].rows();
+  }
+
+  return out;
+}
+
+void bayesmix::append_by_row(Eigen::MatrixXd *a, const Eigen::MatrixXd &b) {
+  if (!a->rows()) {
+    *a = b;
+  } else if (!b.rows()) {
```

Comment:
This syntax for checking equality to 0 is really ugly... And it's also not consistent with the nicer one used in the other function :stuck_out_tongue: